### PR TITLE
Replace helm-broker etcd sub-chart with EtcdCluster

### PR DESCRIPTION
**Pull request by mszostok**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- This PRs replace Helm Broker etcd sub-chart with EtcdCluster CR

### CODE REVIEW

File: resources/core/charts/helm-broker/charts/etcd/templates/_helpers.tpl [hunk 1]
```diff
@@ -1,32 +1,10 @@
 {{/* vim: set filetype=mustache: */}}
-{{/*
-Expand the name of the chart.
-*/}}
-{{- define "etcd.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
-
-{{/*
+ {{/*
```
Comment (PK85) [inline resources/core/charts/helm-broker/charts/etcd/templates/_helpers.tpl]: remove that file, same as we did in service-catalog/etcd, hardoce a name as we did in https://github.com/kyma-project/kyma/blob/6606b10f49e9ca5d3125629da31a463b2465b8c5/resources/core/charts/service-catalog/charts/etcd/templates/etcd-cluster-cr.yaml

File: resources/core/charts/helm-broker/charts/etcd/templates/etcd-cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: "etcd.database.coreos.com/v1beta2"
+kind: "EtcdCluster"
+metadata:
+  name: {{ template "etcd.fullname" . }}
```
Comment (PK85) [inline resources/core/charts/helm-broker/charts/etcd/templates/etcd-cluster.yaml]: hardcode name to core-helm-broker-etcd

File: resources/core/charts/helm-broker/charts/etcd/Chart.yaml [hunk 3]
```diff
@@ -1,10 +1,3 @@
...
 name: etcd
-home: https://github.com/coreos/etcd
-version: 0.4.0
-appVersion: 3.2.14
 description: Distributed reliable key-value store for the most critical data of a distributed system.
-icon: https://raw.githubusercontent.com/coreos/etcd/master/logos/etcd-horizontal-color.png
-sources:
-- https://github.com/coreos/etcd
-
+version: 0.0.1
```
Comment (PK85) [inline resources/core/charts/helm-broker/charts/etcd/Chart.yaml]: could you try to use etcd from service catalog? I think we should reduce number of etcds to one for SC specifc components.

Comment (mszostok) [reply resources/core/charts/helm-broker/charts/etcd/Chart.yaml]: I've tried to achieve this with the `requirements.yaml` but it's not easy and the base chart still needs to be copied into sub-chart. I didn't find any other way how to do that.

File: resources/core/charts/helm-broker/charts/etcd/templates/_helpers.tpl (hunk 1 above)
Comment (mszostok) [reply resources/core/charts/helm-broker/charts/etcd/templates/_helpers.tpl]: but in this PR https://github.com/kyma-project/kyma/pull/92 I also changed the way how the etcd is installed for SC

**Pull request merged**